src/main/MQ2Spawns.cpp

```diff
@@ -81,195 +72,464 @@ static bool dataNamingSpawn(const char* szIndex, MQTypeVar& Ret)
 	return false;
 }
 
-enum eCaptionColor
-{
-	CC_PC = 0,
-	CC_PCConColor = 1,
-	CC_PCPVPTeamColor = 2,
-	CC_PCRaidColor = 3,
-	CC_PCClassColor = 4,
-	CC_PCGroupColor = 5,
-	CC_PCTrader = 6,
-	CC_NPC = 7,
-	CC_NPCConColor = 8,
-	CC_NPCClassColor = 9,
-	CC_NPCMerchant = 10,
-	CC_NPCBanker = 11,
-	CC_NPCAssist = 12,
-	CC_NPCMark = 13,
-	CC_PetNPC = 14,
-	CC_PetPC = 15,
-	CC_PetConColor = 16,
-	CC_PetClassColor = 17,
-	CC_Corpse = 18,
-	CC_CorpseClassColor = 19,
-};
-
-struct CAPTIONCOLOR
+static void ConvertCR(std::string& text)
 {
-	char* szName;
-	char* szDescription;
-	bool       Enabled;
-	bool       ToggleOnly;
-	DWORD      Color;
-};
+	size_t in = 0, out = 0;
+	for (; in < text.size(); in++, out++)
+	{
+		if (text[in] == '\\' && in < text.size() - 1 && text[in + 1] == 'n') {
+			text[out] = '\n';
+			in++;
+		}
+		else if (in != out)
+			text[out] = text[in];
+	}
+	text[out] = 0;
```

Comment:
What's the intent of this that resize and the above in text[out] = text[in] dont' already cover?